File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/TemplateDB/TemplateRow.ts
﻿import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface TemplateRow {
    Id?: string;
    Title?: string;
    TemplateFile?: string;
    IsDefault?: boolean;
    CreatedDate?: string;
    CreatorUserName?: string;
    ModifiedDate?: string;
    ModifiedUserName?: string;
    IsActive?: number;
}

export abstract class TemplateRow {
    static readonly idProperty = 'Id';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'Title';
    static readonly localTextPrefix = 'TemplateDB.Template';
    static readonly lookupKey = 'TemplateDB.Template';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<TemplateRow>('TemplateDB.Template') }
    static async getLookupAsync() { return getLookupAsync<TemplateRow>('TemplateDB.Template') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<TemplateRow>();
}

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/TemplateDB/Template/TemplateRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace CorrespondenceSystem.TemplateDB;

[ConnectionKey("CorrespondenceSystem"), Module("TemplateDB"), TableName("Template")]
[DisplayName("Template"), InstanceName("Template")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript]
public sealed class TemplateRow : Row<TemplateRow.RowFields>, IIdRow, INameRow, IIsActiveRow, ILoggingRow
{
    [DisplayName("Id"), PrimaryKey, NotNull, IdProperty]
    public Guid? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Title"), Size(100), QuickSearch, NameProperty]
    public string Title { get => fields.Title[this]; set => fields.Title[this] = value; }

    [DisplayName("Template File"), Size(200),FileUploadEditor]
    public string TemplateFile { get => fields.TemplateFile[this]; set => fields.TemplateFile[this] = value; }

    [DisplayName("Is Default")]
    public bool? IsDefault { get => fields.IsDefault[this]; set => fields.IsDefault[this] = value; }

    [DisplayName("Created Date")]
    public DateTime? CreatedDate { get => fields.CreatedDate[this]; set => fields.CreatedDate[this] = value; }

    [DisplayName("Creator User Name"), Size(200)]
    public string CreatorUserName { get => fields.CreatorUserName[this]; set => fields.CreatorUserName[this] = value; }

    [DisplayName("Modified Date")]
    public DateTime? ModifiedDate { get => fields.ModifiedDate[this]; set => fields.ModifiedDate[this] = value; }

    [DisplayName("Modified User Name"), Size(200)]
    public string ModifiedUserName { get => fields.ModifiedUserName[this]; set => fields.ModifiedUserName[this] = value; }

    [DisplayName("Is Active")]
    public short? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    public Int16Field IsActiveField => fields.IsActive;

    public Field UpdateUserIdField => fields.ModifiedUserName;

    public DateTimeField UpdateDateField => fields.ModifiedDate;

    public Field InsertUserIdField => fields.CreatorUserName;

    public DateTimeField InsertDateField => fields.CreatedDate;

    public class RowFields : RowFieldsBase
    {
        public GuidField Id;
        public StringField Title;
        public StringField TemplateFile;
        public BooleanField IsDefault;
        public DateTimeField CreatedDate;
        public StringField CreatorUserName;
        public DateTimeField ModifiedDate;
        public StringField ModifiedUserName;
        public Int16Field IsActive;

    }
}